tools/userchroot.nix

```diff
@@ -0,0 +1,60 @@
+{ pkgs ? import <nixpkgs> {} }:
+
+with pkgs;
+
+let
+  tensorflow-c = pkgs.stdenv.mkDerivation {
+
+    name = "tensorflow-c";
+
+    src = fetchurl {
+      url = "https://storage.googleapis.com/tensorflow/libtensorflow/libtensorflow-cpu-linux-x86_64-1.0.0.tar.gz";
+      sha256 = "1kmss9wdri5ibd19b4ymh2dvajw5cy34bxzxlaa5i60hiwpppljy";
+    };
+
+    buildCommand = ''
+      . $stdenv/setup
+      mkdir -pv $out
+      cat $src | tar -C $out  -xz
+    '';
+  };
+
+  fhs = pkgs.buildFHSUserEnv {
+    name = "fhs";
+    targetPkgs = pkgs:
+      with pkgs.python3Packages;
+      with pkgs.haskellPackages;
+      with pkgs;
+      [
+      pkgs.snappy
+      pkgs.snappy.dev
+      pkgs.protobuf3_2
+      stack
```

Comment:
In theory NixOS doesn't decreas versions of packages, so the build rule shouldn't break if it is a part of main <nixpkgs> tree. Currently it is not the case so some asserts are indeed requiered (added).